src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -632,27 +656,54 @@ private Result<?, UpdateError> handleAddedTrip(
     //
     // Create added trip
     //
-
-    Route route = null;
-    if (tripDescriptor.hasRouteId()) {
-      // Try to find route
-      route = getRouteForRouteId(new FeedScopedId(tripId.getFeedId(), tripDescriptor.getRouteId()));
-    }
-
-    if (route == null) {
-      // Create new Route
-      // Use route id of trip descriptor if available
+    // Create dummy agency for added trips
+    Agency dummyAgency = Agency
+      .of(new FeedScopedId(tripId.getFeedId(), "Dummy"))
+      .withName("Dummy")
+      .withTimezone("Europe/Paris")
+      .build();
+    Route route;
+    if (tripDescriptor.hasExtension(MfdzRealtimeExtensions.tripDescriptor)) {
       FeedScopedId id = tripDescriptor.hasRouteId()
         ? new FeedScopedId(tripId.getFeedId(), tripDescriptor.getRouteId())
         : tripId;
 
       var builder = Route.of(id);
```

Comment:
This is a good suggestion and I just tried it out.

However, the problem is that these ADDED routes are not added to the transit index so it's actually quite hard to figure out if it already exits.

Nevertheless, since the route is added to trip pattern which _is_ cached it is still the same instance when there are two consecutive updates. I've also added a test to check that this is the case.

Reply:
Mmm good question, it reveals an existing potential issue when your trip times matches a pattern of another added trip times on another route and isn't properly added with the right route reference as a result. For car pool, if two users provide a similar trop between two city, only one of the user route will be created with two trip times. If there's important information in the route to identify and contact the driver that might be an issue.

Similarly for transit, if two added trips, initially for two different routes/bus lines, share the same set of stops, they might both be displayed as on the same bus line. I feel the cache should be depending on the route id too.

Also the route not being added to the graph index could be problematic if you want to get the route details by route id later on, I feel this would be necessary.

Reply:
I actually found a way to add the route the graph index. It was rather easy: bcc40963da1fa7d1d5630a403a9b7bda1e40f7db

Reply:
I just looked at the on commit above and found two issues - this is why we need to refactor the TransitModel. Of cause you can make the index for route a concurrent map, but by injecting the implementaion/editor into the Snapshot you exposes the Snapshot for threading issues. This adds another stone to the task of refactoring the transit model, but I am not sure if we have another choice...

Reply:
If you have a suggestion I'm happy to implement it.

BTW, the Siri updater does the same thing: https://github.com/opentripplanner/OpenTripPlanner/blob/124ced049b11d1d17cdd6481e0a60a204f90cbd2/src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java#L563